spec/models/user_spec.rb

```diff
@@ -0,0 +1,23 @@
+require "rails_helper"
+require "faker"
+
+describe User, type: :model do
+  let(:board) {FactoryBot.create :board}
+  let(:user) {FactoryBot.create :user}
+  let(:user2) {FactoryBot.create :user}
+  let(:user_board) {FactoryBot.create :user_board, user: user, board: board}
+
+  describe "#join_board" do
+    before {user.join_board board, Faker::Number.between(from: 0, to: 1)}
+    it "join success" do
+      expect(user.join_boards).to include board
+    end
+  end
+
+  describe "#is_leader?" do
+    before {user_board}
```

Comment:
before nó chạy ntn e ?

Reply:
before sẽ chạy trước khi 1 example trong phần describe này chạy ạ, tương tự với after thì sẽ chạy sau ạ

Reply:
1 example là khối nào trong đoạn code này
giả sửa có nhiều khối exxample thì nó chạy 1 lần hay nhiều lần

Reply:
1 example là 1 khối it ạ, với mỗi 1 it thì nó sẽ chạy 1 lần ạ

Reply:
Ready